src/clojure_tictactoe/cli/input/input_getter.clj

```diff
@@ -27,17 +27,26 @@
   (get-user-input)
   )
 
-(defn get-player-choice []
+(defn get-player-choice [board-length]
   (println (str "Which space would you like to mark?"))
-  (if-let [choice (parse-move-input (get-user-input))]
+  (if-let [choice (parse-move-input (get-user-input) board-length)]
     choice
     (do (println "Sorry, looks like that's not possible, try again?")
-        (recur))))
+        (recur board-length))))
 
 (defn get-player-move [board]
+  (let [move (get-player-choice (count board))]
+    (if (board/space-is-open? move board)
+      move
+      (do (println "Sorry, that looks taken, try again")
+          (recur board)))))
+
+(defn get-game-type [options]
+  (println instructions-printer/game-choice-message)
+  (instructions-printer/stringify-options options)
   (loop []
-    (let [move (get-player-choice)]
-      (if (board/space-is-open? move board)
-        move
-        (do (println "Sorry, that looks taken, try again")
-            (recur))))))
+    (println instructions-printer/game-choice-request)
+    (let [choice-limit (+ (count options) 1)]
+      (if-let [choice (parse-move-input (get-user-input) choice-limit)]
```

Comment:
I think you need to rename `parse-move-input` if you're going to use it to parse game type input as well as move input.

I'm wondering if you could just pass the list of choices (either the game types or the board) directly to this input parsing function and have it calculate what the limit should be instead of doing the `(+ (count options) 1)` in this function. I'm also wondering if that same input parsing function can subtract 1 before returning so you don't have to do that in the next line either, though maybe not if your board starts with 0.

Reply:
Yeah, I was thinking about that. I think that's something I might change in the future since it makes sense for the functions and I think as far as ux goes, starting with 1 instead of 0 is probably better. Thanks!